src/tx/builder/index.ts

```diff
@@ -0,0 +1,743 @@
+import { AeAmountFormats } from './../../utils/amount-formatter'
+import { EncodedData, EncodingType } from './../../utils/encoder'
+import BigNumber from 'bignumber.js'
+import { decode as rlpDecode, encode as rlpEncode, NestedUint8Array } from 'rlp'
+import { AE_AMOUNT_FORMATS, formatAmount } from '../../utils/amount-formatter'
+import { hash } from '../../utils/crypto'
+import { Field } from './field-types'
+
+import {
+  DEFAULT_FEE,
+  FIELD_TYPES,
+  OBJECT_ID_TX_TYPE,
+  TxField,
+  TxParams,
+  TxType,
+  TX_DESERIALIZATION_SCHEMA,
+  TX_FEE_BASE_GAS,
+  TX_FEE_OTHER_GAS,
+  TX_SERIALIZATION_SCHEMA,
+  TX_TYPE,
+  VSN
+} from './schema'
+import {
+  readInt,
+  readId,
+  readPointers,
+  writeId,
+  writeInt,
+  buildPointers,
+  encode,
+  decode,
+  Pointer
+} from './helpers'
+import { toBytes } from '../../utils/bytes'
+import MPTree, { MPTreeBinary } from '../../utils/mptree'
+import { InvalidTxParamsError, SchemaNotFoundError } from '../../utils/errors'
+
+/**
+ * JavaScript-based Transaction builder
+ * @module @aeternity/aepp-sdk/es/tx/builder
+ * @example import { calculateFee } from '@aeternity/aepp-sdk'
+ */
+
+/**
+ * @interface TxOptionsCommon
+ * These options are common and can be provided to every tx-type.
+ */
+interface TxOptionsCommon {
```

Comment:
I'm not sure that we will have a general options at the end, I would combine options of child method calls instead